test/TestCaseReader.cpp

```diff
@@ -137,6 +144,54 @@ pair<SourceMap, size_t> TestCaseReader::parseSourcesAndSettingsWithLineNumber(is
 				if (sources.count(currentSourceName))
 					BOOST_THROW_EXCEPTION(runtime_error("Multiple definitions of test source \"" + currentSourceName + "\"."));
 			}
+			else if (boost::algorithm::starts_with(line, externalSourceDelimiterStart) && boost::algorithm::ends_with(line, sourceDelimiterEnd))
+			{
+				string externalSourcePath = boost::trim_copy(line.substr(
+					externalSourceDelimiterStart.size(),
+					line.size() - sourceDelimiterEnd.size() - externalSourceDelimiterStart.size()
+				));
+				string externalSourceName = externalSourcePath;
+
+				// Does the external source define a remapping?
+				size_t remappingPos = externalSourcePath.find('=');
+				if (remappingPos != string::npos)
+				{
+					externalSourceName = boost::trim_copy(externalSourcePath.substr(0, remappingPos));
+					externalSourcePath = boost::trim_copy(externalSourcePath.substr(remappingPos + 1));
+				}
+
+				string externalSourceContent;
+				fs::path parent_path(m_filename);
+				parent_path = parent_path.normalize().remove_filename();
+
+				fs::path path = parent_path / fs::path(externalSourcePath);
```

Comment:
There are so many paths in here that calling it just `path` is very confusing.
```suggestion
				fs::path externalSourceFullPath = parent_path / fs::path(externalSourcePath);
```